packages/flutter_tools/lib/src/ios/application_package.dart

```diff
@@ -106,26 +101,26 @@ abstract class IOSApp extends ApplicationPackage {
 
   /// Directory used by ios-deploy to store incremental installation metadata for
   /// faster second installs.
-  Directory get appDeltaDirectory;
+  Directory? get appDeltaDirectory;
 }
 
 class BuildableIOSApp extends IOSApp {
-  BuildableIOSApp(this.project, String projectBundleId, String hostAppBundleName)
+  BuildableIOSApp(this.project, String? projectBundleId, String? hostAppBundleName)
     : _hostAppBundleName = hostAppBundleName,
-      super(projectBundleId: projectBundleId);
+      super(projectBundleId: projectBundleId ?? 'com.example.Runner');
```

Comment:
What are the mechanics of filling in `com.example.Runner'` here? If this field was actually missing are we going to fail somewhere else? should we just fail earlier?

Reply:
I have a better idea.  `projectBundleId` can be made nonnull if `BuildableIOSApp.fromProject` can return a nullable project, and its only callsite `IOSApp.fromIosProject` handles a null project.

That probably didn't make sense, diff (with test) incoming...

Reply:
See https://github.com/flutter/flutter/pull/83956/commits/609c217da0170518a6409358b1d8f304edeaf239